JS/cart.js

```diff
@@ -147,99 +162,113 @@ function createCart2(article) {
+
+            let img_cart_div = document.createElement('div');
+            img_cart_div.appendChild(img);
+            img_cart_div.className = "img_cart_empty";
+
+            let warning = document.createTextNode("Il n'y a aucun article dans le panier");
+            let warning_div = document.createElement('div');
+            warning_div.className = "name_cart";
+            warning_div.appendChild(warning);
+            warning_div.style = "margin: 0 auto; text-align : center;";
+
+            cart_div.appendChild(img_cart_div);
+            cart_div.appendChild(warning_div);
+            document.getElementById("nb_object_in_cart").textContent = "Hum...";
+            document.getElementById("total_to_pay").textContent = "";
+            document.getElementById("nb_article_in_cart").textContent = 0;
+            return cart_div;
         }
     }
 
+    document.getElementById("overlay_alert_btn_no").onclick = function () {
+        document.getElementById("overlay_alert").style.display = "none";
+    }
+
     /*
     * We update the sessionStorage with the new quantity.
     * We re-calculate the price of the article according to the new quantity
     */
     box_select_plus.onclick = function () {
-        var intermediaire = document.getElementById('quantity_id' + article.id).value;
-        document.getElementById('quantity_id' + article.id).value = parseInt(intermediaire) + 1;
-        price_div.removeChild(price);
-        total_price = total_price - parseFloat(price.textContent);
-        price = document.createTextNode((article.price * (parseInt(intermediaire) + 1)).toFixed(2) + "€");
-        price_div.appendChild(price);
-        total_price = total_price + parseFloat(price.textContent);
+        let elem = document.getElementById('quantity_id' + article.id);
+        elem.value = parseInt(elem.value) + 1;        
+        total_price = total_price - parseFloat(price.textContent) + article.price * elem.value;
+        price.nodeValue = (article.price * elem.value).toFixed(2) + "€";
         document.getElementById("total_to_pay").textContent = "Total : " + total_price.toFixed(2) + " €";
-        var quantite = document.getElementById('quantity_id' + article.id).value;
-        var old_data_saved = JSON.parse(sessionStorage.getItem('articleToCart2'));
-        var tab = [];
-
-        for (i = 0; i < Object.keys(old_data_saved).length; i++) {
-            if (old_data_saved[i].id == article.id) {
-                old_data_saved[i].quantity = parseInt(quantite);
-            }
-            tab.push(old_data_saved[i]);
-        }
-        sessionStorage.setItem('articleToCart2', JSON.stringify(tab));
+        update_json_file(article, elem);
     };
 
+ /*
+there is a lot of call to JSON.parse and JSON.stringify, without counting the numberof parseInt|Float. 
+The cart should get a class with an internal storage dumped when necessary this will remove many excessive call to those functions. 
```

Comment:
I am just putting an emphasis on this : 
The cart should get it's own class, it will make things much easier to understand and more efficient.